Re-organizing and editing text input docs (#918)
hashicorp/design-system: website/docs/components/form/text-input/partials/code/how-to-use.md

## How to use this component

!!! Info

> Depending on how you're processing the user input upon submission (eg. server-side via form `POST` or client-side using JavaScript) you will need to provide a `name` attribute or a custom `ID` attribute to the field. Since the decision on how to process the input data is left to the consumers, we omit these specific arguments in the examples, for sake of simplicity.

Reviewer: ```suggestion
Depending on how you’re processing the user input upon submission (eg. server-side via form `POST` or client-side using JavaScript) you will need to provide a `name` attribute or a custom `ID` attribute to the field. Since the decision on how to process the input data is left to the consumers, we omit these specific arguments in the examples, for sake of simplicity.
```

</Hds::Form::TextInput::Field>
```

#### Validation

> To indicate a field is invalid, declare it's invalid using the `@isInvalid` argument and provide an error message using the `Error` contextual component.

Reviewer: ```suggestion
To indicate a field is invalid, declare that it’s invalid by using the `@isInvalid` argument and provide an error message using the `Error` contextual component.
```

</Hds::Form::TextInput::Field>
```

#### Custom control ID

> In case it's necessary to have custom ID for the control, instead of the one automatically generated by the component (eg. because it needs to be referenced in the code for other reasons), you need to pass a `@id` argument to the field.

Reviewer: ```suggestion
If it’s necessary to have custom ID for the control, instead of the one automatically generated by the component (i.e., because it needs to be referenced in the code for other reasons), pass the `@id` argument to the field.
```

</Hds::Form::TextInput::Field>
```

> #### Extra "aria-describedby"

Reviewer: ```suggestion
#### Additional `aria-describedby`
```
